File: Data Structures & Algorithms/non-overlapping-intervals/submission-1.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        removed = 0
        intervals.sort(key = lambda x : x[1])
        prev = intervals[0][1]

        for start, end in intervals[1:]:
            if prev > start:
                removed += 1
                prev = min(end, prev)
            else:
                prev = end
        return removed

File: Data Structures & Algorithms/edit-distance/submission-1.cpp
class Solution {
public:
    int minDistance(string word1, string word2) {
        int n = word1.size();
        int m = word2.size();
        vector<vector<int>> cp(n+1, vector<int>(m+1, 0));
        for (int i = 0; i <= n; i++) {
            cp[i][0] = i;
        }
        for (int j = 0; j <= m; j++) {
            cp[0][j] = j;
        }

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= m; j++) {
                if (word1[i-1] == word2[j-1]) {
                    cp[i][j] = cp[i-1][j-1];
                }
                else {
                    cp[i][j] = min({cp[i-1][j], cp[i][j-1], cp[i-1][j-1]}) + 1;
                }
            }
        }
        return cp[n][m];
    }
};
